rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1015,24 +1038,64 @@ private String getTaskResultValueAsStringOrExceptionStackTrace(TaskResult taskRe
     }
 
     @Override
-    public byte[] serializedValueOfTaskResult(String sessionId, String jobId, String taskname) throws Throwable {
+    public byte[] serializedValueOfTaskResult(String sessionId, String jobId, String taskname) throws RestException {
         Scheduler s = checkAccess(sessionId, PATH_JOBS + jobId + PATH_TASKS + taskname + "/result/serializedvalue");
-        TaskResult tr = s.getTaskResult(jobId, taskname);
-        tr = PAFuture.getFutureValue(tr);
-        return tr.getSerializedValue();
+        try {
+            TaskResult tr = s.getTaskResult(jobId, taskname);
+            tr = PAFuture.getFutureValue(tr);
+            return tr.getSerializedValue();
+        } catch (SchedulerException e) {
+            throw RestException.wrapExceptionToRest(e);
+        }
+    }
+
+    @Override
+    public Response downloadTaskResult(String sessionId, String jobId, String taskname, String destination)
+            throws RestException {
+        byte[] serializedValue = serializedValueOfTaskResult(sessionId, jobId, taskname);
+        Map<String, String> resultMetadata = metadataOfTaskResult(sessionId, jobId, taskname);
+        if (resultMetadata == null) {
+            resultMetadata = new HashMap<>();
+        }
+        if (destination == null) {
+            destination = DESTINATION_BROWSER;
+        }
+
+        Response.ResponseBuilder builder = Response.ok().entity(new ByteArrayInputStream(serializedValue));
+
+        if (resultMetadata.containsKey(METADATA_CONTENT_TYPE)) {
+            builder = builder.header(HttpHeaders.CONTENT_TYPE, resultMetadata.get(METADATA_CONTENT_TYPE));
```

Comment:
@fviale what is no content-type is provided ? Could you throw an exception with an explicit and clear error message ?

Reply:
no content-type is standard, it is when resultMetada is not filled. In that case I should return application/octet-stream (I will add this)

Reply:
alternatively, I can return a byte stream of the result toString and set the content type to text/plain, it could make it easier to handle then.

Reply:
Yes, I meant when resultMetada is not filled. Should we make it mandatory to fill it or return a bytearray by default ?

Reply:
I updated the PR to handle everything related to result. Now this endpoint can be called directly without using any other endpoints (task failed, no metadata, etc).